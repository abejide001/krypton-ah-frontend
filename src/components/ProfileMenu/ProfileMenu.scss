.profileMenu {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
  align-content: center;

  .profileMenuIcon {
    height: 1.3em;
    align-self: center;
  }
}

.profileMenu :hover {
  cursor: pointer;
}

.count {
  background-color: black;
  color: white;
  margin-left: 0.3rem;
}

.profileMenuText {
  margin-top: 0.2rem;
}

.hidden {
  display: none;
}
