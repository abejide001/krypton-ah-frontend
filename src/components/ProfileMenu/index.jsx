import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './ProfileMenu.scss';

function ProfileMenu(props) {
  const { text, to, count, menuItem } = props;
  if (text) {
    return (
      <Link className="menu-item profileMenu" to={to}>
        {menuItem}
        <div className="profileMenuText">
          {text}
          <span className={count ? 'count' : 'hidden'}>{count}</span>
        </div>
      </Link>
    );
  }
  return <div className="menu-item profileMenu">{menuItem}</div>;
}

ProfileMenu.defaultProps = {
  text: '',
  count: '',
  to: ''
};

ProfileMenu.propTypes = {
  text: PropTypes.string,
  count: PropTypes.string,
  menuItem: PropTypes.object.isRequired,
  to: PropTypes.string
};

export default ProfileMenu;
