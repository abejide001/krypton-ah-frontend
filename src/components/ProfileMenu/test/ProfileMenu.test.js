import React from 'react';
import { shallow } from 'enzyme';
import ProfileMenu from '..';

describe('ProfileMenu', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ProfileMenu menuItem={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
