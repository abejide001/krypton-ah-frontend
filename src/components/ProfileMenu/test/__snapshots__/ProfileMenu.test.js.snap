// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu should match snapshot 1`] = `
<div
  className="menu-item profileMenu"
/>
`;
